src/test/java/io/antmedia/test/StreamFetcherUnitTest.java

```diff
@@ -251,9 +264,9 @@ public void testRestartPeriodStreamFetcher() {
 			when(streamFetcher.getStream()).thenReturn(stream);
 			when(streamFetcher.isStreamAlive()).thenReturn(true);
 			when(streamFetcher.getCameraError()).thenReturn(new Result(true));
-			
+
 			StreamFetcherFactory factory = mock(StreamFetcherFactory.class);
-			when(factory.make(stream, appScope)).thenReturn(streamFetcher);
+			when(factory.make(stream, appScope, scheduler)).thenReturn(streamFetcher);
```

Comment:
Instead of using mock, you can make the "make" method public static and directly call StreamFetcherFactory .make(stream, appScope, scheduler). in this case you need to mock "make" method parameters.Or you can remove Factory class and make the "make" method  a method of StreamFetcherManager so that you can mock it.

Reply:
Could you please implement it for me :)